### ISSUE TITLE
[Feature Request] Show me my TODOs (no-warning-comments)

### ISSUE THREAD

**Issue by micky2be** [eslint/eslint#4409]
When getting output about the no-warning-comments rule, it would be nice to show the actual comment associated with the terms (TODO, FIXME, ...)

That could become very useful for developers

The code:
```js
[...]
// TODO: add the logic for...
function () {
[...]
```
could output:
```bash
/index.js
  20:1  warning  Todo comment: add the logic for... no-warning-comments
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
I'm not sure it's a great idea. While I think it can be very helpful, comments can span multiple lines and have special characters in them, which might break some of the formatters. @eslint/eslint-team anyone feels this is worth it to start escaping every error message inside formatters?

**Comment by IanVS** [comment]
What exactly would the benefit be here?  If you don't want those types of comments in your code, isn't it enough to know that one exists (and it's type) in a file at a particular location?

**Comment by ilyavolodin** [comment]
If it's set as a warning, you can use it as a quick task list. Some editors, like Visual Studio, will automatically pull them out and let you see all of the TODO comments from the whole codebase.

**Comment by IanVS** [comment]
Hm.  That's not really the purpose of this rule though, the point of turning it on is to prevent such comments.  I guess I'm still :-1: but I don't feel strongly about it.

**Comment by micky2be** [comment]
The way I see it for this rule is not that you wrote "TODO" in your code but that you actually have an unfinished task. So let's informed the developer.
That's kinda the goal of TODOs.

Leaving as it is just feel like, "hey! you wrote '// TODO:' in your code and we don't allow it".
If the real goal is to avoid this kind of comments, well people are gonna write comments without putting TODO as a prefix.

Anyway, that was just a suggestion. I can use a secondary tool on the side.

**Comment by platinumazure** [comment]
Not sure that's the purpose of a linting tool honestly. Lint tools are for
detecting "lint" (that is, undesirable patterns) in code and helping you
remove them. If you want to allow TODO comments, then they're not really
lint in your project and having ESLint report on them goes beyond the
purpose of this project.

If you want something that will enumerate your TODOs and you can't find a
satisfactory existing tool, you could possibly write your own tool built on
top of espree and estraverse which spits those comments out. But I don't
think ESLint will fit your needs.

**Comment by micky2be** [comment]
Understood. No problem about that philosophy.
